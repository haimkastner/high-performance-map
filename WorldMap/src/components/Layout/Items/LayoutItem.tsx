import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";

interface Props {
  children: JSX.Element[] | JSX.Element;
  opacity: number;
  color: string;
  width: string;
  height: string;
  top?: number;
  bottom?: number;
  left: number;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    layout: {
      position: "fixed",
      zIndex: 1000
    },
    color: {
      position: "absolute",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      zIndex: 999
    },
    item: {
      zIndex: 1000,
      position: "relative"
    }
  })
);

export function LayoutItem(props: Props) {
  const classes = useStyles();

  const layoutStyle = {
    width: props.width,
    height: props.height,
    top: props.top ?? "",
    bottom: props.bottom ?? "",
    left: props.left
  };

  const colorStyle = {
    backgroundColor: props.color,
    opacity: props.opacity
  };

  return (
    <Box className={classes.layout} style={layoutStyle}>
      <Box className={classes.item}>{props.children}</Box>
      <Box className={classes.color} style={colorStyle} />
    </Box>
  );
}
