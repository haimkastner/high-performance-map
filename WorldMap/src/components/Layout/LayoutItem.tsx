import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";

interface Props {
  children: JSX.Element[] | JSX.Element;
  opacity: number;
  color: string;
  width: string;
  height: string;
  top?: number | string;
  bottom?: number | string;
  left?: number | string;
  right?: number | string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    layout: {
      position: "fixed",
      zIndex: 1000
    },
    color: {
      position: "absolute",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      zIndex: 999,
      backgroundColor: "ffjhdifuhdsfu"
    },
    item: {
      zIndex: 1000,
      position: "relative"
    }
  })
);

export function LayoutItem(props: Props) {
  const classes = useStyles();

  const layoutStyle = {
    width: props.width,
    height: props.height,
    top: props.top ?? "",
    bottom: props.bottom ?? "",
    left: props.left ?? "",
    right: props.right ?? ""
  };

  const colorStyle = {
    backgroundColor: props.color,
    opacity: props.opacity
  };

  const itemStyle = {
    width: "100%",
    height: "100%"
  };

  return (
    <Box className={classes.layout} style={layoutStyle}>
      <Box className={classes.item} style={itemStyle}>
        {" "}
        {props.children}
      </Box>
      <Box className={classes.color} style={colorStyle} />
    </Box>
  );
}
