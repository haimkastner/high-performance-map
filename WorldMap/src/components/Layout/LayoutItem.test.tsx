import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { LayoutItem } from './LayoutItem';
import color from '@material-ui/core/colors/yellow';
import { shallow } from 'enzyme';
import { Container } from '@material-ui/core';

describe('LayoutItem Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LayoutItem children={<div id="test"/>} opacity={1} color={"blue"} width={"3"} height={"3"}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<LayoutItem children={<div id="test"/>} opacity={1} color={"blue"} width={"3"} height={"3"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('LayoutItem Component Logic', () => {
      const childC = <div id="tester"/>;
      const opacityC = 1;
      const colorC = "blue";
      const widthC = "3";
      const heightC = "3";

      const wrapper = shallow(<LayoutItem children={childC} opacity={opacityC} color={colorC} width={widthC} height={heightC} />);
      const content = wrapper.find('#content-box');
      const container = wrapper.find('#container');
      const background = wrapper.find('#background');

      expect(content.contains(childC)).toEqual(true);
      expect(background.props().style?.opacity).toEqual(opacityC);
      expect(background.props().style?.backgroundColor).toEqual(colorC);
      expect(container.props().style?.width).toEqual(widthC);
      expect(container.props().style?.height).toEqual(heightC);

  });
});
