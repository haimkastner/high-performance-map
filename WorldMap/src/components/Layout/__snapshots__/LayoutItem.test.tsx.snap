// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutItem Component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-9 makeStyles-layout-1"
  id="container"
  style={
    Object {
      "bottom": "",
      "height": "3",
      "left": "",
      "right": "",
      "top": "",
      "width": "3",
    }
  }
>
  <div
    className="MuiBox-root MuiBox-root-10 makeStyles-color-2"
    id="background"
    style={
      Object {
        "backgroundColor": "blue",
        "opacity": 1,
      }
    }
  />
  <div
    className="MuiBox-root MuiBox-root-11 makeStyles-item-3"
    id="content-box"
    style={Object {}}
  >
    <div
      id="test"
    />
  </div>
</div>
`;
